tests/test_codex_permission_prompt_notification.py

```
@@ -0,0 +1,271 @@
+    cli_path: str, root: Path
+) -> None:
+    """A one-way write would let the hook exit before the app processed the
+    notification, so the CLI must await the app's acknowledgement. With the
+    fake delaying its OK by 0.5s, a fire-and-forget regression returns
+    almost instantly; the awaited transport cannot."""
+    delay = 0.5
+    stdout, frames, elapsed = run_feed_hook_capture(
+        cli_path, root / "cmux-ack.sock", "PermissionRequest", raw_response_delay=delay
+    )
+    if stdout != {}:
+        raise AssertionError(f"PermissionRequest must stay non-blocking: {stdout!r}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)